### ISSUE TITLE
calcite-inline-editable throws error if no calcite-input is slotted

### ISSUE THREAD

**Issue by driskull** [Esri/calcite-components#3780]
### Actual Behavior

An inline-editable component throws an error if no input is slotted.

```
p-2e29554c.js:5 TypeError: Cannot set properties of null (setting 'disabled')
    at componentWillLoad (p-8ebc2d51.entry.js:5)
    at K (p-2e29554c.js:5)
    at V (p-2e29554c.js:5)
    at Array.<anonymous> (p-2e29554c.js:5)
    at we (p-2e29554c.js:5)
    at be (p-2e29554c.js:5) undefined
```

### Expected Behavior

There should be a visual error displayed instead of just a JS error that is pretty vague.

### Reproduction Steps and Sample

Add an inline-editable and do not slot a calcite-input. 

OR

slot the input after initialization.

```
<calcite-inline-editable></calcite-inline-editable>
```

You should be able to slot the element at any time. The only check now is at `componentWillLoad` which is not ideal.

_Sample_:

1. https://codepen.io/driskull/pen/jOGGwNV?editors=1000

### Relevant Info

_Reproducible version_: `@esri/calcite-components@1.0.0-<version>`

- [ ] CDN
- [ ] NPM package

Regression? _Last working version_: `@esri/calcite-components@1.0.0-<version>`

**Comment by driskull** [comment]
@jcfranco @macandcheese @eriklharper @asangma  What do you think should happen when a `calcite-inline-editable` doesn't have a `calcite-input` slotted?

- Should we show nothing?
- Should there be an error displayed?
- Should we render a basic `calcite-input` using the default slot?

**Comment by driskull** [comment]
I think we should also discuss this pattern>

```
/** specify the scale of the inline-editable component, defaults to the scale of the wrapped calcite-input or the scale of the closest wrapping component with a set scale */
  @Prop({ reflect: true, mutable: true }) scale?: Scale;
```

It's weird that the scale looks down or up through the DOM.

**Comment by macandcheese** [comment]
This component has always been a bit weird in structure to me - is it possible to consider having “input editable” just be an opt-in mode of input component? 

Obviously a bigger lift but this seems like an outlier as far as usage goes.

**Comment by asangma** [comment]
I agree with @macandcheese here.
Doing that to the `CalciteInput` might also go a long way to enabling our list item components to have an inline-editable UX.

**Comment by driskull** [comment]
@benelan @jcfranco can you guys comment on the proposal and triage a new issue if applicable?

**Comment by eriklharper** [comment]
> @jcfranco @macandcheese @eriklharper @asangma What do you think should happen when a `calcite-inline-editable` doesn't have a `calcite-input` slotted?
> 
> * Should we show nothing?
> * Should there be an error displayed?
> * Should we render a basic `calcite-input` using the default slot?

Option #3 makes the most sense to me.  I'm realizing too that the name of the component should probably include the word `input` in it to communicate than an input will be rendered, but that's just an aside.

**Comment by eriklharper** [comment]
> I think we should also discuss this pattern>
> 
> ```
> /** specify the scale of the inline-editable component, defaults to the scale of the wrapped calcite-input or the scale of the closest wrapping component with a set scale */
>   @Prop({ reflect: true, mutable: true }) scale?: Scale;
> ```
> 
> It's weird that the scale looks down or up through the DOM.

I have this on the agenda to discuss today.

**Comment by eriklharper** [comment]
> This component has always been a bit weird in structure to me - is it possible to consider having “input editable” just be an opt-in mode of input component?
> 
> Obviously a bigger lift but this seems like an outlier as far as usage goes.

I like this idea as well, but I would prefer if we do go this route to break out number functionality into the `calcite-input-number` component before we add editable behavior to CalciteInput.  If others feel the same, I would happily take on the task of moving all the number functionality out of Input into the new InputNumber component.  I've been wanting to do that for a long while especially because there are always lots of bugs related to number (I'm working on one now in fact) and it would be a relief to be able to fix bugs without having to worry about applying logic that doesn't affect text input mode.

**Comment by macandcheese** [comment]
> > This component has always been a bit weird in structure to me - is it possible to consider having “input editable” just be an opt-in mode of input component?
> > Obviously a bigger lift but this seems like an outlier as far as usage goes.
> 
> I like this idea as well, but I would prefer if we do go this route to break out number functionality into the `calcite-input-number` component before we add editable behavior to CalciteInput.

Yep, same with textarea!

**Comment by jcfranco** [comment]
For the immediate issue, I think we should update the component to work regardless of when the input is slotted. 

Unless I'm missing a use case, this situation seems like a user error. We could update the doc to emphasize that slotting an input is required. I believe all examples have an input already. @driskull do you recall where this came from?

> Should we render a basic calcite-input using the default slot?

I initially liked this as well, but reconsidered after thinking about getting the underlying input value. We would need to add a `value` prop (and others?), which pretty much make it a `calcite-input`-lite™. This tells me we should consider the proposed option of enhancing `calcite-input` with this mode. 

On a related note, would the inline-editable pattern apply to all input types? @eriklharper and @macandcheese touched on this, but we'll be splitting input into separate type components, so it'd be good to know how it'll fit into all of those.

> We should also get @alexabreu 's input as well on this idea as he can comment further on the use cases he created this component for.

👍🏼

**Comment by driskull** [comment]
> I initially liked this as well, but reconsidered after thinking about getting the underlying input value. We would need to add a value prop (and others?), which pretty much make it a calcite-input-lite™. This tells me we should consider the proposed option of enhancing calcite-input with this mode.

We can have `calcite-input` that internally renders a `calcite-input-editable` which could also be a standalone component.

That way, we get the best of both worlds... a mode and code separation.

**Comment by eriklharper** [comment]
Yeah if we can somehow encapsulate the inline editable experience in a separate component that we can compose in all of the (eventually one day) separate input components, that would be the best reuse pattern that keeps all the use cases consistent.  Since right now its assumed an input will be slotted, then I imagine we will have to support textarea and others.

**Comment by driskull** [comment]
This pr fixes any errors when an input is not slotted in an inline-editable. https://github.com/Esri/calcite-components/pull/3889

However, it doesn't show any error state if the slotted input is missing.

@eriklharper can you review the PR?

**Comment by jcfranco** [comment]
This got fixed in beta.76 via b00bdf7 and no longer throws an error: https://codepen.io/jcfranco/pen/qBVmJWB?editors=1000. Marking this one as verified! cc @benelan
